quattor/types/metadata.pan

```diff
@@ -14,7 +14,28 @@ type structure_template = {
     "branch" : structure_branch
 };
 
+type structure_os_version = {
+    @{Descriptive form of version, e.g. 7x}
+    "name" : string
+    @{Major version number component, e.g. 7}
+    "major" : long(0..)
+    @{Minor version number component}
+    "minor" ? long(0..)
+};
+
+type structure_os = {
+    @{Distribution name, e.g. sl}
+    "distribution" : string
```

Comment:
It's supposed to be usable, the definitions of operating systems in our template libraries all use abbreviations. I've add validation to match that in `standard/os/version`.